civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -0,0 +1,61 @@
+/* globals angular */
+
+(function (angular) {
+  angular.module('taDocuments', ['civitasks.appDocuments']).controller('ModalController', ['$scope', '$rootScope', '$rootElement', '$log', '$uibModal', 'DocumentService', 'FileService', 'config',
+    function ($scope, $rootScope, $rootElement, $log, $modal, DocumentService, FileService, config) {
+      var vm = {};
+
+      /**
+       * Gets Document for the given document id and
+       * opens Modal with retrived document data
+       *
+       * @param {integr} id
+       * @param {string} role
+       */
+      vm.modalDocument = function (id, role) {
+        DocumentService.get({ id: id })
+          .then(function (data) {
+             if (!data) {
+               throw new Error('Requested Document is not available');
+             }
+
+             !!data && openModalDocument(data[0], role);
```

Comment:
there is no point checking for `!!data` anymore. If `data` is not present, you are throwing an exception